src/ray/core_worker/transport/actor_scheduling_queue.cc

```diff
@@ -146,26 +171,38 @@ void ActorSchedulingQueue::ScheduleRequests() {
     pending_actor_tasks_.erase(head);
   }
 
-  // Process as many in-order requests as we can.
-  while (!pending_actor_tasks_.empty() &&
-         pending_actor_tasks_.begin()->first == next_seq_no_ &&
-         pending_actor_tasks_.begin()->second.CanExecute()) {
-    auto head = pending_actor_tasks_.begin();
-    auto request = head->second;
-    auto task_id = head->second.TaskID();
+  // Process as many retry requests as we can.
+  // Retry requests do not respect sequence number ordering, so we execute them as soon as
+  // they are ready to execute.
+  auto retry_iter = pending_retry_actor_tasks_.begin();
+  while (retry_iter != pending_retry_actor_tasks_.end()) {
+    auto &request = *retry_iter;
+    if (!request.CanExecute()) {
```

Comment:
`CanExecute` should really just be `DependenciesResolved`... so much unnecessary indirection...

let's clean up later

Reply:
we seem to check that dependencies are resolved on submitter and executor side (haven't looked that closely tho)